pkg/edgeview/src/crypto.go

```diff
@@ -163,3 +334,232 @@ func encryptVarInit(jdata types.EvjwtInfo) {
+		Type:  "RSA PUBLIC KEY",
+		Bytes: x509EncodedPubKey,
+	}
+	pemBytes := pem.EncodeToMemory(pemBlock)
+
+	return pemBytes, nil
+}
+
+// getEdgeviewSSHPublicKeys reads multiple SSH public keys from a single string
+func getEdgeviewSSHPublicKeys() []string {
+	// Split the keysString by newline to get individual keys
+	var validKeys []string
+	configitems := getConfigItems()
+	items, ok := configitems.GlobalSettings[types.EdgeviewPublicKeys]
+	if ok && items.StrValue != "" {
+		keys := strings.Split(items.StrValue, "\n")
+
+		// Remove any empty strings from the slice
+		for _, key := range keys {
+			trimmedKey := strings.TrimSpace(key)
+			if trimmedKey != "" {
+				validKeys = append(validKeys, trimmedKey)
+			}
+		}
+	}
+	return validKeys
+}
+
+// getEdgeviewClientPrivateKey - Edgeview client mounts the private-key path inside the container
+func getEdgeviewClientPrivateKey() string {
+	privateKeyPath := "/clientauth/privatekey"
+	privateKeyBytes, err := os.ReadFile(privateKeyPath)
+	if err != nil {
+		fmt.Printf("Failed to read private key file: %v\n", err)
+		return ""
+	}
+	return string(privateKeyBytes)
+}
+
+// ExtractCommentFromSSHPublicKey extracts the comment from an SSH public key string
+// to be used in logging the user-info of the Edgeview command
+func ExtractCommentFromSSHPublicKey(publicSSHKey string) (string, error) {
+	// Parse the SSH public key
+	_, comment, _, _, err := ssh.ParseAuthorizedKey([]byte(publicSSHKey))
+	if err != nil {
+		return "", fmt.Errorf("failed to parse SSH public key: %v", err)
+	}
+	return comment, nil
+}
+
+// signWithECPrivateKey signs the message using the provided EC private key
+func signWithECPrivateKey(msg []byte, privateKey *ecdsa.PrivateKey) ([]byte, error) {
+	hashed := sha256.Sum256(msg)
+	r, s, err := ecdsa.Sign(rand.Reader, privateKey, hashed[:])
+	if err != nil {
+		return nil, err
+	}
+
+	// Concatenate r and s to form the signature
+	signature := append(r.Bytes(), s.Bytes()...)
+	return signature, nil
+}
+
+func signClientCertAuthenData(msg []byte, ecPrivateKey string) []byte {
+	jmsg := envelopeMsg{
+		Message: msg,
+	}
+
+	privateKey, err := loadPrivateKey(ecPrivateKey)
+	if err != nil {
+		log.Errorf("failed to load private key: %v", err)
+		return nil
+	}
+
+	var signature []byte
+	switch key := privateKey.(type) {
+	case *rsa.PrivateKey:
+		signature, err = signWithRSAPrivateKey(jmsg.Message, key)
+		if err != nil {
+			log.Errorf("failed to sign message: %v", err)
+			return nil
+		}
+	case *ecdsa.PrivateKey:
+		signature, err = signWithECPrivateKey(jmsg.Message, key)
+		if err != nil {
+			log.Errorf("failed to sign message: %v", err)
+			return nil
+		}
+	default:
+		log.Errorf("Unknown private key type: %T", privateKey)
+		return nil
+	}
+
+	jmsg.Signature = signature
+
+	jdata, err := json.Marshal(jmsg)
+	if err != nil {
+		log.Errorf("json marshal error: %v", err)
+		return nil
+	}
+	return jdata
+}
+
+// loadPrivateKey loads an EC private key from a PEM file
```

Comment:
Looks like it can load ECC or RSA private keys. Can you update the comment?

Reply:
Ok.